File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;
import java.util.List;

public interface BrandService {

    /**
    * @Description: 查询所欲的品牌数据
    * @params      * @param null
    * @return
    * @exception
    * @date        2019/7/3 9:48
    */
    public List<TbBrand> findAll();

    /**
     * @Description: 分页查询
     * @params      * pageNum 当前页数 pageSize 页码大小
     * @return
     * @exception
     * @date        2019/7/3 9:47
     */
    public PageResult findPage(int pageNum,int pageSize);


    /**
     * @Description: 分页查询
     * @params      * pageNum 当前页数 pageSize 页码大小
     * @return
     * @exception
     * @date        2019/7/3 9:47
     */
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);



    /**
    * @Description: 增加品牌列表
    * @params      * @param null
    * @return      
    * @exception   
    * @date        2019/7/3 10:54
    */
    public void add(TbBrand brand);

    /**
    * @Description: 批量删除品牌
    * @params      * @param null
    * @return
    * @exception
    * @date        2019/7/3 14:17
    */
    public void delete(Long[] ids);

    /**
    * @Description: 查询品牌
    * @params      * @param null
    * @return
    * @exception
    * @date        2019/7/4 23:32
    */
    public TbBrand findOne(Long id);

    /**
    * @Description: 更新品牌
    * @params      * @param null
    * @return
    * @exception
    * @date        2019/7/4 23:32
    */
    public void update(TbBrand brand);





}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//商品品牌 前端服务层
app.service('brandService',function ($http) {
    this.findAll= function(){
        return $http.get('../brand/findAll.do');
    };
    this.findPage= function (pageNum, pageSize) {
        return $http.get('../brand/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize);
    };
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?'+'id='+id);
    }
    this.add =function (entity) {
        return  $http.post('../brand/add.do', entity);
    }
    this.update =function (entity) {
        return  $http.post('../brand/update.do', entity);
    }
    this.dele =function (ids) {
        return  $http.get('../brand/delete.do?ids=' + ids);
    }

    this.search =function (pageNum,pageSize,searchEntity) {
        return $http.post('../brand/search.do?pageNum='+pageNum +'&pageSize='+pageSize, searchEntity)

    }
});